flowzone.yml

```
@@ -1773,6 +1855,46 @@ jobs:
             | jq -r '.stacks[].name' | jq -Rcn '[inputs'])"
           echo "deploy=${stacks}" >> $GITHUB_OUTPUT
 
+  is_terraform:
+    name: Is Terraform
+    runs-on: ${{ fromJSON(inputs.runs_on) }}
+    timeout-minutes: ${{ fromJSON(inputs.jobs_timeout_minutes) }}
+    needs:
+      - versioned_source
+    if: |
+      !failure() && !cancelled() &&
+      needs.versioned_source.result == 'success' &&
+      inputs.terraform_projects != ''
+
+    <<: *customWorkingDirectory
+
+    outputs:
+      terraform: ${{ steps.validate_json.outputs.enabled }}
+      projects: ${{ steps.terraform_projects.outputs.deploy }}
+
+    steps:
+      - name: Validate Terraform input
+        id: validate_json
+        run: |
+          if echo '${{ inputs.terraform_projects }}' | jq -r .projects
+          then
+            if [[ '${{ github.event.pull_request.head.repo.full_name }}' != '${{ github.repository }}' ]]; then
+              echo '::warning:: Terraform projects are skipped for external contributions.'
+              exit 0
```

```suggestion
              echo "enabled=false" >> $GITHUB_OUTPUT
              exit 0
```
Either set it to false in both failed conditions, or never set it to false and only check for true conditions. Not a mix of both.

---

I don't follow. The current logic would not set step outputs if the request is coming from a fork.

---

I know, but there is another condition with `echo '::warning::invalid JSON or no CloudFormation stacks?'` where we are setting it explicitly to false.

All I'm saying is be consistent. Either set it to false, or don't set it. But be consistent.